chainer/training/updater.py

```diff
@@ -128,7 +128,8 @@ def __init__(self, iterator, optimizer, converter=convert.concat_examples,
         self._optimizers = optimizer
 
         if device is not None and device >= 0:
-            optimizer['main'].target.to_gpu(device)
+            for opt in self._optimizers.values():
```

Comment:
How about using `six.itervalues(self._optimizers)` instead of `self._optimizers.values()`?